src/Chapter3.hs

```diff
@@ -1005,6 +1105,13 @@ properties using typeclasses, but they are different data types in the end.
 Implement data types and typeclasses, describing such a battle between two
 contestants, and write a function that decides the outcome of a fight!
 -}
+data Defence = Defence Int
+data KnightF  = KnightF Health Attack Defence
+data MonsterF = MonsterF Health Attack
+{-
+  I need to learn more to finish these task, i tried but i dont have clear ideas for now. 
+  I will come back after chapter 4.
```

Comment:
No worries! This is quite big tasks with lots of concepts together 🙂 

In addition to `Knight` and `Monster` types, I would recommend creating separate types for `MonsterAction` and `KnightAction` to represent possible actions for each type correspondingly.

Since each data type (`Monster` and `Knight`) can be fighters, I can suggest creating a typeclass of the following shape:

```haskell
class Fighter a where
    ... you need to think of possible methods here ...
```

So you can them implement a recursive function using this typeclass:

```haskell

data FightResult a b
    = FirstWins a
    | SecondWins b

fight :: (Fighter a, Fighter b) => a -> b -> FightResult a b
```

I hope this clarifies the task a bit more and gives some ideas on how to proceed with it :slightly_smiling_face: